azure-blob-storage/index.js

```diff
@@ -0,0 +1,73 @@
+const azure = require('azure-storage');
+const getOpt = require('node-getopt');
+const moment = require('moment');
+
+let NagiosPlugin = require('nagios-plugin');
+let plugin = new NagiosPlugin({
+    shortName: 'azure_blobs'
+});
+
+getOptInstance = getOpt
+.create([['s', 'storageAccount=<STRING>', 'Storage account name']
+    , ['c', 'container=<STRING>', 'Container name']
+    , ['t', 'token=<STRING>', 'SAS token with read access to the desired container']
+    , ['p', 'filePrefix=<STRING>', 'File prefix']
+    , ['d', 'daysBack=<STRING>', 'Days backs to search blobs']
+    , ['h', 'help', 'display this help']])
+.bindHelp()
+.setHelp('Usage: node index.js\n[[OPTIONS]]')
+
+const args = getOptInstance.parseSystem();
+
+function verify_args(_getopt, args) {
+    for (arg of args) {
+        if (!arg) {
+            console.log(_getopt.getHelp());
+            process.exit(1);
+        }
+    }
+}
+
+const promsifiedListBlobsSegmentedWithPrefix = async (blobService, container, filePrefix, continuationToken) => {
+    return new Promise((resolve, reject) => {
+        blobService.listBlobsSegmentedWithPrefix(args.options.container, args.options.filePrefix, continuationToken, (err, result) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(result)
+            }
+        })
+    })
+}
+
+async function run() {    
+    verify_args(getOptInstance, [args.options.storageAccount, args.options.container, args.options.token, args.options.filePrefix, args.options.daysBack])
+
+    var blobUri = 'https://' + args.options.storageAccount + '.blob.core.windows.net';
+    let blobService = azure.createBlobServiceWithSas(blobUri, args.options.token);
+
+    let entities = [];
+    let first = true;
+    let continuationToken;
+    while (continuationToken || first) {
+        first = false;
+        let result = await promsifiedListBlobsSegmentedWithPrefix(blobService, args.options.container, args.options.filePrefix, continuationToken);
+        entities = entities.concat(result.entries);
+        continuationToken = result.continuationToken;
+    }      
+    
+    entities = entities.filter(entity => moment(entity.lastModified) > moment().subtract(args.options.daysBack, 'd'));
```

Comment:
Wouldn't if be more efficient to do the filter in the loop? If we have blob with many files, it will load all of them into the memory...

Reply:
It loads only the metadata, not a big deal, but I'll change it :)